### PR TITLE
Test with Swift 5.3 on Linux

### PR DESCRIPTION
**Pull request by MaxDesiatov**
I'm not sure if you'd like to test with both Swift 5.2 and 5.3 on Linux. As you're using a cache, I think it would be easier to test just with a single version, as it doesn't make sense to share that cache. Let me know if you'd prefer to test both versions after all with separate cache keys.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
Of course, `SwiftSyntax parser library isn't compatible` 😞

**Comment by MaxDesiatov** [comment]
This requires changes to `swift-doc` dependencies, I'll update those first...

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,35 +10,43 @@ jobs:
...
         env:
-          DEVELOPER_DIR: /Applications/Xcode_11.4.app/Contents/Developer
+          DEVELOPER_DIR: /Applications/Xcode_${{ matrix.xcode }}.app/Contents/Developer
 
   linux:
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        swift: ["5.3", "5.2"]
```
Comment (MaxDesiatov) [inline .github/workflows/ci.yml:36]: I think this won't work, we'll need two different versions of SwiftSyntax used for each Swift version. The fact that SwiftSyntax can't be linked in statically (or redistributed with `swift-doc` itself) is a deal-breaker 😑

**Comment by mattt** [comment]
@MaxDesiatov Thanks for getting a head start on this! A lot of the work I'd been wanting to do on `swift-doc` had been blocked on general availability of Swift 5.3, so I'm really excited for this to finally be coming in place.

I just pushed a few commits to test Swift 5.2 and 5.3 on CI for Linux and macOS (I'd like to keep support for 5.2 in the short term, to give folks some time to upgrade).

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mattt) [reply .github/workflows/ci.yml:36]: We should be able to work around that, no? Could we use Swift version-specific manifests like `Package@swift-5.2.swift`?

**Comment by MaxDesiatov** [comment]
Thanks! This now depends on https://github.com/NSHipster/SwiftSyntaxHighlighter/pull/8

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MaxDesiatov) [reply .github/workflows/ci.yml:36]: Spot on, I forgot about that trick, I'll add it to NSHipster/SwiftSyntaxHighlighter#8.

**Review by mattt — APPROVED** [review]

**Comment by mattt** [comment]
Thanks for your help with this, @MaxDesiatov! Glad to have this blocker out of the way so that we can target Swift 5.3 for future releases.

**Comment by MaxDesiatov** [comment]
Thanks for finishing the PR!

**Pull request merged**